### PR TITLE
 Add Integration event support to Configurator

### LINKED ISSUE #120: Add Integration event support to Configurator

### PR DESCRIPTION
**Pull request by NikitaEgorov**
fix #120

### CODE REVIEW

**Review by aleshchev — APPROVED** [review]

File: src/Framework.Configurator/Handlers/CreateBusinessRoleHandler.cs [hunk 1]
```diff
@@ -47,12 +59,12 @@ public List<Guid> OperationIds
         {
             get;
             set;
-        }
+        } = null!;
```
Comment (ivan-luxoft) [inline src/Framework.Configurator/Handlers/CreateBusinessRoleHandler.cs:62]: maybe `default!` is better?

File: src/Framework.Configurator/Handlers/CreatePrincipalHandler.cs [hunk 2]
```diff
@@ -1,28 +1,31 @@
...
 
 using Microsoft.AspNetCore.Http;
 
 namespace Framework.Configurator.Handlers;
 
-public class CreatePrincipalHandler : BaseWriteHandler, ICreatePrincipalHandler
-
+public record CreatePrincipalHandler(
+        IAuthorizationRepositoryFactory<Principal> AuthorizationRepositoryFactory,
+        IConfiguratorIntegrationEvents? ConfiguratorIntegrationEvents = null) : BaseWriteHandler, ICreatePrincipalHandler
```
Comment (ivan-luxoft) [inline src/Framework.Configurator/Handlers/CreatePrincipalHandler.cs:15]: how about a default/empty implementation (null object pattern)?

Comment (ivan-luxoft) [inline src/Framework.Configurator/Handlers/CreatePrincipalHandler.cs:15]: how about a default/empty implementation (null object pattern)?

Comment (NikitaEgorov) [reply src/Framework.Configurator/Handlers/CreatePrincipalHandler.cs:15]: No thanks :)

File: src/Framework.Configurator/Handlers/CreateBusinessRoleHandler.cs (hunk 1 above)
Comment (NikitaEgorov) [reply src/Framework.Configurator/Handlers/CreateBusinessRoleHandler.cs]: Fixed

File: src/Framework.Configurator/Handlers/CreatePrincipalHandler.cs (hunk 2 above)
Comment (ivan-luxoft) [reply src/Framework.Configurator/Handlers/CreatePrincipalHandler.cs:15]: I mean use the non-nullable argument for service, and register built-in service - it even exists - [SampleConfiguratorIntegrationEvents.cs](https://github.com/Luxoft/BSSFramework/blob/64f88dfc01e45be554958b515cdac5a5ebb382d4/src/_SampleSystem/SampleSystem.WebApiCore/Services/SampleConfiguratorIntegrationEvents.cs)`

so we can remove the conditional check for executing the method
```
if (this.ConfiguratorIntegrationEvents != null)
{
    await this.ConfiguratorIntegrationEvents.PrincipalCreatedAsync(principal, cancellationToken);
}
```

**Review by ivan-luxoft — APPROVED** [review]

**Pull request merged**